components/user-mgt/org.wso2.carbon.um.ws.service/src/main/java/org/wso2/carbon/um/ws/service/UserStoreManagerService.java

```diff
@@ -57,6 +59,13 @@ public void addUser(String userName, String credential, String[] roleList, Claim
 
     public void setUserClaimValues(String userName, ClaimValue[] claims, String profileName)
             throws UserStoreException {
+        // only allow alphanumeric characters for profile
+        Pattern regex = Pattern.compile("^[a-zA-Z0-9_]*$");
+        Matcher matcher = regex.matcher(profileName);
+
+        if (!matcher.matches()) {
+            throw new UserStoreException("Profile name cannot contain special characters other than underscore!");
+        }
         getUserStoreManager().setUserClaimValues(userName, convertClaimValueToMap(claims),
```

Comment:
We need to check whether this is an already created profile; not profile "name" is valid...

So i guess we need to have UserOperationEventListener in profile mgt component to check above condition.